main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala

```diff
@@ -1026,6 +1029,26 @@ object SemanticTokensProvider {
     case _ => false
   }
 
+  /**
+    * Splits every multiline token in the given list into single-line tokens.
+    */
+  private def splitToken(tokens: List[SemanticToken]): List[SemanticToken] = {
+    tokens.flatMap {
```

Comment:
flatMap can have complexity O(nˆ2) and this list is very long.

Perhaps we should append to a ArrayBuffer and then convert that to a List at the end